### ISSUE TITLE
Gnome shell crashes often

### ISSUE THREAD

**Issue by arnogon** [pop-os/shell#647]
**(1) Issue/Bug Description:**
Gnome shell crashes frequently (multiple times a day).

**(2) Steps to reproduce (if you know):**
Workflow 1: open Firefox, enter stack mode, open the app Planner (Flatpak)
```
Oct 20 09:28:20 pc gnome-shell[2569]: pop-shell:  [INFO] bench load_desktop_files: 69 ms elapsed
Oct 20 09:28:21 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Activating via systemd: service name='org.freedesktop.Flatpak' unit='flatpak-session-helper.servic>
Oct 20 09:28:21 pc systemd[2407]: Starting flatpak session helper...
Oct 20 09:28:21 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Successfully activated service 'org.freedesktop.Flatpak'
Oct 20 09:28:21 pc systemd[2407]: Started flatpak session helper.
Oct 20 09:28:21 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Activating via systemd: service name='org.freedesktop.portal.Documents' unit='xdg-document-portal.>
Oct 20 09:28:21 pc systemd[2407]: Starting flatpak document portal service...
Oct 20 09:28:21 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Successfully activated service 'org.freedesktop.portal.Documents'
Oct 20 09:28:21 pc systemd[2407]: Started flatpak document portal service.
Oct 20 09:28:21 pc systemd[2407]: Started app-flatpak-com.github.alainm23.planner-4829.scope.
Oct 20 09:28:22 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.ser>
Oct 20 09:28:22 pc systemd[2407]: Starting Portal service...
Oct 20 09:28:22 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gtk' unit='xdg-desktop-p>
Oct 20 09:28:22 pc systemd[2407]: Starting Portal service (GTK+/GNOME implementation)...
Oct 20 09:28:22 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Successfully activated service 'org.freedesktop.impl.portal.desktop.gtk'
Oct 20 09:28:22 pc systemd[2407]: Started Portal service (GTK+/GNOME implementation).
Oct 20 09:28:22 pc pipewire[2430]: [W][000000121.135046][module-protocol-native.c:386 client_new()] server 0x56391fc67c40: no peersec: Protocol not available
Oct 20 09:28:22 pc dbus-daemon[2435]: [session uid=1000 pid=2435] Successfully activated service 'org.freedesktop.portal.Desktop'
Oct 20 09:28:22 pc systemd[2407]: Started Portal service.
Oct 20 09:28:22 pc gnome-shell[2569]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:28:22 pc gnome-shell[2569]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:28:22 pc gnome-shell[2569]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:28:22 pc gnome-shell[2569]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:28:23 pc gnome-shell[2569]: Error in freeze/thaw accounting
Oct 20 09:28:23 pc gnome-shell[2569]: == Stack trace for context 0x56209a51d1c0 ==
Oct 20 09:28:23 pc gnome-shell[2569]: #0   56209cb1c428 i   resource:///org/gnome/shell/ui/windowManager.js:1384 (3287b61ced80 @ 589)
Oct 20 09:28:23 pc gnome-shell[2569]: #1   7ffe984bb520 b   self-hosted:850 (3287b61f9d80 @ 454)
Oct 20 09:28:23 pc xdg-desktop-por[4863]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc evolution-alarm[2749]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc gsd-media-keys[2734]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc com.github.alai[4854]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc megasync[2819]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc unknown[2759]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc gsd-color[2724]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc gsd-power[2736]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc systemd[2407]: org.gnome.SettingsDaemon.Wacom.service: Main process exited, code=exited, status=1/FAILURE
Oct 20 09:28:23 pc gsd-keyboard[2733]: Error reading events from display: Broken pipe
Oct 20 09:28:23 pc systemd[2407]: org.gnome.SettingsDaemon.MediaKeys.service: Main process exited, code=exited, status=1/FAILURE
Oct 20 09:28:23 pc systemd[2407]: org.gnome.Shell@wayland.service: Main process exited, code=killed, status=5/TRAP
Oct 20 09:28:23 pc systemd[2407]: org.gnome.SettingsDaemon.Color.service: Main process exited, code=exited, status=1/FAILURE
Oct 20 09:28:23 pc systemd[2407]: org.gnome.SettingsDaemon.Keyboard.service: Main process exited, code=exited, status=1/FAILURE
Oct 20 09:28:23 pc systemd[2407]: org.gnome.SettingsDaemon.Power.service: Main process exited, code=exited, status=1/FAILURE
Oct 20 09:28:23 pc systemd[2407]: xdg-desktop-portal-gtk.service: Main process exited, code=exited, status=1/FAILURE
Oct 20 09:28:23 pc systemd[2407]: xdg-desktop-portal-gtk.service: Failed with result 'exit-code'.
```

Workflow 2: open terminal, enter stack mode, open dconf Editor
```
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor ClutterActor is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WindowClone is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overviewGroup is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overview is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_overviewControls_ControlsManager is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overview-group is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_overviewControls_ThumbnailsSlider is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_ThumbnailsBox is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WorkspaceThumbnail is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor ClutterActor is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WindowClone is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overviewGroup is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overview is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_overviewControls_ControlsManager is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overview-group is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_overviewControls_ThumbnailsSlider is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_ThumbnailsBox is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WorkspaceThumbnail is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor ClutterActor is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WindowClone is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overviewGroup is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overview is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_overviewControls_ControlsManager is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor overview-group is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_overviewControls_ThumbnailsSlider is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_ThumbnailsBox is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WorkspaceThumbnail is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor ClutterActor is on because it needs an allocation.
Oct 20 09:32:48 pc gnome-shell[5437]: Can't update stage views actor Gjs_ui_workspaceThumbnail_WindowClone is on because it needs an allocation.
Oct 20 09:32:50 pc gnome-shell[5437]: pop-shell:  [INFO] bench load_desktop_files: 51 ms elapsed
Oct 20 09:32:52 pc dbus-daemon[4963]: [session uid=1000 pid=4963] Activating service name='ca.desrt.dconf-editor' requested by ':1.18' (uid=1000 pid=5437 comm="/usr>
Oct 20 09:32:52 pc dbus-daemon[4963]: [session uid=1000 pid=4963] Successfully activated service 'ca.desrt.dconf-editor'
Oct 20 09:32:52 pc gnome-shell[5437]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:32:52 pc gnome-shell[5437]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:32:52 pc gnome-shell[5437]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:32:52 pc gnome-shell[5437]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Oct 20 09:32:52 pc gnome-shell[5437]: Error in freeze/thaw accounting
Oct 20 09:32:52 pc gnome-shell[5437]: == Stack trace for context 0x560ac431e110 ==
Oct 20 09:32:52 pc gnome-shell[5437]: #0   560ac4e2ab38 i   resource:///org/gnome/shell/ui/windowManager.js:1384 (1247f30ced80 @ 589)
Oct 20 09:32:52 pc gnome-shell[5437]: #1   7ffc1de63770 b   self-hosted:850 (1247f30f9d80 @ 454)
Oct 20 09:32:52 pc gnome-shell[5437]: #2   7ffc1de649d0 b   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:127 (1ba9daa81a10 @ 48>
Oct 20 09:32:52 pc gnome-shell[5437]: #3   7ffc1de64a90 b   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/executor.js:29 (1ba9daaa8ab0 @ 74)
Oct 20 09:32:52 pc gnome-terminal-[6058]: Error reading events from display: Broken pipe
```

These two workflows are not the only ones that can cause the crash, but the ones I know how to reproduce.

**(3) Expected behavior:**
No crashes.

**(4) Distribution (run `cat /etc/os-release`):**
 Debian testing

**(5) Gnome Shell version:**
3.38.0 (Wayland)

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
a11d3c34db01987bb716b8b127b2b889130a4fc1

**(7) Where was Pop Shell installed from:**
Git repository

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**
1 x 1080p

**(9) Other Installed/Enabled Extensions:**
Many, but bug is reproducible even with all the other extensions disabled.

**(10) Other Notes:**
Crashes mostly happen when I use stack mode, but I have experienced crashes even when consciously avoiding using stack mode. I will update the issue as I figure out more ways to reproduce the crashes.

**Comment by mmstick** [comment]
I have not seen any crashes on Pop. Probably a GNOME 3.38 or Wayland thing

**Comment by arnogon** [comment]
> I have not seen any crashes on Pop. Probably a GNOME 3.38 or Wayland thing

So I just tried to reproduce the bug in X11. Gnome shell still crashed, but instead of logging me out, it just froze for a second, then opened the new app on the side (I was in stack mode but it was reset after the crash). This is probably just because of how Wayland and X11 handled the crash differently?

Also if I close the new app and redo the workflows again it won't crash the second time.

Workflow 1
```
Oct 20 10:33:32 pc systemd[2445]: Started Application launched by gnome-shell.
Oct 20 10:33:35 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating service name='org.gnome.ChromeGnomeShell' requested by ':1.100' (uid=1000 pid=4350 comm>
Oct 20 10:33:35 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.gnome.ChromeGnomeShell'
Oct 20 10:33:38 pc gnome-shell[2729]: pop-shell:  [WARN] Window(2,0) does not have an actor, and therefore cannot be moved
Oct 20 10:33:39 pc gnome-shell[2729]: pop-shell:  [WARN] Window(1,1) does not have an actor, and therefore cannot be moved
Oct 20 10:33:39 pc gnome-shell[2729]: Source ID 3599 was not found when attempting to remove it
Oct 20 10:33:42 pc tracker-store[2548]: OK
Oct 20 10:33:42 pc systemd[2445]: tracker-store.service: Succeeded.
Oct 20 10:33:57 pc gnome-shell[2729]: pop-shell:  [INFO] bench load_desktop_files: 48 ms elapsed
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating via systemd: service name='org.freedesktop.Flatpak' unit='flatpak-session-helper.servic>
Oct 20 10:34:00 pc systemd[2445]: Starting flatpak session helper...
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.freedesktop.Flatpak'
Oct 20 10:34:00 pc systemd[2445]: Started flatpak session helper.
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating via systemd: service name='org.freedesktop.portal.Documents' unit='xdg-document-portal.>
Oct 20 10:34:00 pc systemd[2445]: Starting flatpak document portal service...
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.freedesktop.portal.Documents'
Oct 20 10:34:00 pc systemd[2445]: Started flatpak document portal service.
Oct 20 10:34:00 pc systemd[2445]: Started app-flatpak-com.github.alainm23.planner-4744.scope.
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.ser>
Oct 20 10:34:00 pc systemd[2445]: Starting Portal service...
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gtk' unit='xdg-desktop-p>
Oct 20 10:34:00 pc systemd[2445]: Starting Portal service (GTK+/GNOME implementation)...
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.freedesktop.impl.portal.desktop.gtk'
Oct 20 10:34:00 pc systemd[2445]: Started Portal service (GTK+/GNOME implementation).
Oct 20 10:34:00 pc pipewire[2468]: [W][000000077.807241][module-protocol-native.c:386 client_new()] server 0x55a63178bc40: no peersec: Protocol not available
Oct 20 10:34:00 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.freedesktop.portal.Desktop'
Oct 20 10:34:00 pc systemd[2445]: Started Portal service.
Oct 20 10:34:00 pc gnome-shell[2729]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x500002f specified for 0x5000007.
Oct 20 10:34:01 pc gnome-shell[2729]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x500002f
Oct 20 10:34:01 pc gnome-shell[2729]: Error in freeze/thaw accounting
Oct 20 10:34:01 pc gnome-shell[2729]: == Stack trace for context 0x562c7e3c12e0 ==
Oct 20 10:34:01 pc gnome-shell[2729]: #0   7ffe6c6021f0 b   resource:///org/gnome/shell/ui/windowManager.js:1384 (3e2a7ced80 @ 589)
Oct 20 10:34:01 pc gnome-shell[2729]: #1   7ffe6c6022d0 b   self-hosted:850 (3e2a7fad80 @ 454)
Oct 20 10:34:01 pc gnome-shell[2729]: #2   562c7f938608 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/window.js:194 (5a03f6a7100 @ 154)
Oct 20 10:34:01 pc gnome-shell[2729]: #3   562c7f938560 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/forest.js:664 (5a03f686330 @ 203)
Oct 20 10:34:01 pc gnome-shell[2729]: #4   562c7f938468 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/forest.js:51 (5a03f686fb0 @ 587)
Oct 20 10:34:01 pc gnome-shell[2729]: #5   562c7f9383a0 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/forest.js:35 (5a03f689060 @ 131)
Oct 20 10:34:01 pc gnome-shell[2729]: #6   562c7f9382c8 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:277 (49dec93b0b0 @ 31)
Oct 20 10:34:01 pc gnome-shell[2729]: #7   562c7f9381c8 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:120 (49dec93b4c0 @ 74>
Oct 20 10:34:01 pc gnome-shell[2729]: #8   562c7f9380c8 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:133 (49dec93b470 @ 14>
Oct 20 10:34:01 pc gnome-shell[2729]: #9   7ffe6c603ec0 b   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:144 (49dec93b420 @ 19>
Oct 20 10:34:01 pc gnome-shell[2729]: #10   7ffe6c604630 b   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:1449 (3fd1c63e1c40 @ >
Oct 20 10:34:01 pc systemd[2445]: org.gnome.Shell@x11.service: Main process exited, code=killed, status=5/TRAP
```

Workflow 2
```
Oct 20 10:39:30 pc systemd[2445]: gnome-terminal-server.service: Succeeded.
Oct 20 10:39:33 pc gnome-shell[4824]: showing
Oct 20 10:39:36 pc systemd[2445]: Started Application launched by gsd-media-keys.
Oct 20 10:39:36 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating via systemd: service name='org.gnome.Terminal' unit='gnome-terminal-server.service' req>
Oct 20 10:39:36 pc systemd[2445]: Starting GNOME Terminal Server...
Oct 20 10:39:36 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.gnome.Terminal'
Oct 20 10:39:36 pc systemd[2445]: Started GNOME Terminal Server.
Oct 20 10:39:36 pc systemd[2445]: Started VTE child process 5431 launched by gnome-terminal-server process 5421.
Oct 20 10:39:36 pc systemd[2445]: app-gnome-gnome\x2dterminal-5416.scope: Succeeded.
Oct 20 10:39:36 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating via systemd: service name='org.freedesktop.Tracker1' unit='tracker-store.service' reque>
Oct 20 10:39:36 pc systemd[2445]: Starting Tracker metadata database store and lookup manager...
Oct 20 10:39:36 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'org.freedesktop.Tracker1'
Oct 20 10:39:36 pc systemd[2445]: Started Tracker metadata database store and lookup manager.
Oct 20 10:39:38 pc gnome-shell[4824]: pop-shell:  [INFO] bench load_desktop_files: 34 ms elapsed
Oct 20 10:39:40 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Activating service name='ca.desrt.dconf-editor' requested by ':1.120' (uid=1000 pid=4824 comm="/us>
Oct 20 10:39:40 pc dbus-daemon[2473]: [session uid=1000 pid=2473] Successfully activated service 'ca.desrt.dconf-editor'
Oct 20 10:39:40 pc gnome-shell[4824]: Error in freeze/thaw accounting
Oct 20 10:39:40 pc gnome-shell[4824]: == Stack trace for context 0x55ddccceb2e0 ==
Oct 20 10:39:40 pc gnome-shell[4824]: #0   7ffc7aff39b0 b   resource:///org/gnome/shell/ui/windowManager.js:1384 (2175472ced80 @ 589)
Oct 20 10:39:40 pc gnome-shell[4824]: #1   7ffc7aff3a90 b   self-hosted:850 (2175472fad80 @ 454)
Oct 20 10:39:40 pc gnome-shell[4824]: #2   55ddcd564358 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/window.js:194 (4da56ca7100 @ 154)
Oct 20 10:39:40 pc gnome-shell[4824]: #3   55ddcd5642b0 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/forest.js:664 (4da56c86330 @ 203)
Oct 20 10:39:40 pc gnome-shell[4824]: #4   7ffc7aff5600 b   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/forest.js:51 (4da56c86fb0 @ 587)
Oct 20 10:39:40 pc gnome-shell[4824]: #5   55ddcd5641e8 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/forest.js:35 (4da56c89060 @ 131)
Oct 20 10:39:40 pc gnome-shell[4824]: #6   55ddcd564110 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:277 (2ce0eda3b0b0 @ 3>
Oct 20 10:39:40 pc gnome-shell[4824]: #7   55ddcd564010 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:120 (2ce0eda3b4c0 @ 7>
Oct 20 10:39:40 pc gnome-shell[4824]: #8   55ddcd563f00 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/auto_tiler.js:147 (2ce0eda3b420 @ 2>
Oct 20 10:39:40 pc gnome-shell[4824]: #9   55ddcd563e58 i   /home/user/.local/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:1449 (a74a5fe1c40 @ 97)
Oct 20 10:39:40 pc systemd[2445]: org.gnome.Shell@x11.service: Main process exited, code=killed, status=5/TRAP
Oct 20 10:39:40 pc systemd[2445]: org.gnome.Shell@x11.service: Failed with result 'signal'.
Oct 20 10:39:40 pc systemd[2445]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 2.
Oct 20 10:39:40 pc systemd[2445]: Stopped GNOME Shell on X11.
Oct 20 10:39:40 pc systemd[2445]: Starting GNOME Shell on X11...
Oct 20 10:39:40 pc gnome-shell[5536]: Enabling experimental feature 'scale-monitor-framebuffer'
Oct 20 10:39:40 pc /usr/libexec/gdm-x-session[2560]: (II) modeset(0): EDID vendor "BOE", prod id 1979
Oct 20 10:39:40 pc /usr/libexec/gdm-x-session[2560]: (II) modeset(0): Printing DDC gathered Modelines:
Oct 20 10:39:40 pc /usr/libexec/gdm-x-session[2560]: (II) modeset(0): Modeline "1920x1080"x0.0  141.37  1920 1968 2000 2142  1080 1083 1089 1100 +hsync -vsync (66.0>
Oct 20 10:39:40 pc /usr/libexec/gdm-x-session[2560]: (II) modeset(0): Modeline "1920x1080"x0.0  113.09  1920 1968 2000 2142  1080 1083 1089 1100 +hsync -vsync (52.8>
Oct 20 10:39:40 pc gnome-shell[5536]: Failed to read monitors config file '/home/user/.config/monitors.xml': Logical monitor scales must be identical
Oct 20 10:39:40 pc gsd-media-keys[2872]: Failed to grab accelerators: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.gnome.Shell” on o>
Oct 20 10:39:41 pc gnome-shell[5536]: Some code accessed the property 'CredentialManager' on the module 'credentialManager'. That property was defined with 'let' or>
Oct 20 10:39:41 pc gnome-shell[5536]: Getting parental controls for user 1000
Oct 20 10:39:41 pc gnome-shell[5536]: Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
Oct 20 10:39:41 pc gnome-shell[5536]: Will monitor session 2
```

So looks like this is a Gnome 3.38 problem?

**Comment by mmstick** [comment]
Yeah this is a bug in GNOME 3.38. I wonder if there's a bug report. This particular stack trace points to that:

```
Oct 20 10:34:00 pc gnome-shell[2729]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x500002f specified for 0x5000007.
Oct 20 10:34:01 pc gnome-shell[2729]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x500002f
Oct 20 10:34:01 pc gnome-shell[2729]: Error in freeze/thaw accounting
Oct 20 10:34:01 pc gnome-shell[2729]: == Stack trace for context 0x562c7e3c12e0 ==
Oct 20 10:34:01 pc gnome-shell[2729]: #0   7ffe6c6021f0 b   resource:///org/gnome/shell/ui/windowManager.js:1384 (3e2a7ced80 @ 589)
```

**Comment by mmstick** [comment]
But as soon as I get around to testing pop-shell on 3.38, I may find a workaround for it.

**Comment by ronaaron** [comment]
I just tried tiling on 20.10 (never used it previously). Running Wayland in hybrid mode on an oryxp6.

It seems quite unstable, logging me out and closing all applications without prior notice.  I've got 5 workspaces, fixed, at the moment but only had:  ws1: Konsole X2;  ws2: Vivaldi;  ws3: Thunderbird and Liferea

It's erratic enough that I couldn't determine why it was logging me out. I've returned to non-tiling mode and things are as stable as I expect.

**Comment by mmstick** [comment]
May be fixed by #347 ?

**Comment by jawadalkassim** [comment]
happens to me in 20.10 Wayland too, Gnome keeps crashes and log out . forced me to switch back to X11

**Comment by mmstick** [comment]
Bug fixed in GNOME 3.38.2
